### PR TITLE
Extend FRR BGP Prometheus metrics collection

### PR DESCRIPTION
**Pull request by liornoy**
This commit extends the number of metrics we collect about BGP sessions
using FRR.

The following additional metrics are extracted from the FRR's vtysh
"show bgp neighbors" command.
- OpensSent
- OpensReceived
- NotificationsSent
- UpdatesSent
- UpdatesReceived
- KeepalivesSent
- KeepalivesReceived
- RouteRefreshSent
- TotalSent
- TotalReceived

All of the metrics above are counters of different BGP messages and
they are collected per BGP neighbor.

Note:
The choice of the metric name: "updates_total_received"
was made to keep backward compatibility with the already existing `bgpstats.UpdatesSent.Name`

### CODE REVIEW

**Comment by liornoy** [comment]
Fixed the issue with native failing.
Plus changed the Test Failed message, thanks to @oribon 
Example of the failed test output:
image

File: frr-metrics/collector/bgp_test.go [hunk 1]
```diff
@@ -16,37 +16,91 @@ var (
 	# HELP metallb_bgp_announced_prefixes_total Number of prefixes currently being advertised on the BGP session
 	# TYPE metallb_bgp_announced_prefixes_total gauge
 	metallb_bgp_announced_prefixes_total{peer="{{ .NeighborIP }}"} {{ .AnnouncedPrefixes }}
+	# HELP metallb_bgp_keepalives_received Number of BGP keepalive messages received
```
Comment (fedepaol) [inline frr-metrics/collector/bgp_test.go:19]: nit: can you change the commit subject with the part you are adding the tests for?

File: e2etest/bgptests/bgp.go [hunk 2]
```diff
@@ -272,6 +272,156 @@ var _ = ginkgo.Describe("BGP", func() {
 			framework.ExpectNoError(err)
 		})
 
+		table.DescribeTable("should collect BGP metrics in FRR mode", func(ipFamily ipfamily.Family, poolAddress string, addressTotal int) {
```
[the remark this answers is not in the document]
Comment (liornoy) [reply e2etest/bgptests/bgp.go:275]: done

File: frr-metrics/collector/bgp_test.go (hunk 1 above)
Comment (liornoy) [reply frr-metrics/collector/bgp_test.go:19]: done

**Comment by liornoy** [comment]
@fedepaol Thanks for the review, I fixed the commits titles, and also changed:
Variable name in parse_test.go from `stats` to `expectedStats`

**Review by oribon — APPROVED** [review]
lgtm

**Pull request merged**